invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessLinks/AccessDropdown.js

```diff
@@ -26,15 +28,27 @@ export class AccessDropdown extends Component {
     this.cancellableAction && this.cancellableAction.cancel();
   }
 
+  onSuccess = () =>
+    this.setState({ loading: false, actionSuccess: true, error: undefined });
+
   handleUpdate = async (permission) => {
     const { updateEndpoint } = this.props;
     const data = { permission: permission };
     this.setState({ loading: true, actionSuccess: false });
     this.cancellableAction = withCancel(http.patch(updateEndpoint, data));
 
     try {
+      const requestStartTime = Date.now();
       await this.cancellableAction.promise;
-      this.setState({ loading: false, actionSuccess: true, error: undefined });
+      const requestEndTime = Date.now();
+      const requestDuration = requestEndTime - requestStartTime;
+
+      // prolong loading for better understanding that the action was completed
+      if (requestDuration < MIN_LOADING_DURATION) {
```

Comment:
I think it would improve readability if you use `Promise.all` instead of conditional statements and timeouts

Reply:
I don't see how using `Promise.all()` is going to help get rid of conditional statements and timeouts?

Reply:
if I remember correctly, you could go with
```
await Promise.all([setTimout(MIN_LOADING_DURATION), this.cancellableAction.promise])
this.onSuccess()
```
 so it will just wait until all the promises are fulfilled and you don't need to calculate request duration

Reply:
Looks better indeed! 
Just updated, thank you for the idea 😃